doc/IMPLEMENTATION.md

```diff
@@ -0,0 +1,326 @@
+# Jangouts - Implementation and Working Details
+
+
+## How Janus makes it possible?
+
+![Janus connections](janus.png?raw=true "Janus connections")
+
+Suppose there are `n` participants in a videoroom
+   * All the participants are connected to Janus instance running on the server
+   * For each participant, a session is created (i.e. `n` sessions here for `n`
+    clients).
+   * Each session has a set of `n` different handles for `n` participants. All 
+    sets are different.
+   * Each participant keeps `n` peer connections to Janus (instead of having peer
+    connections directly to the other participant's browser/client). Out of n peer
+    connections, participant uses 1 peer connection for sending it's own content, 
+    and, other n-1 peer connections for consuming other n-1 participants' content. 
+   * Then, Janus forwards the content between these connections.
+
+*Jangouts* exploits this feature of *Janus* to connect clients and provides an
+interactive UI to make communication possible among them.
+
+
+## Some important Components in Jangouts (Brief details)
+
+01. [room.service.js](../src/app/components/room/room.service.js) - It is a 
+service that provides methods to communicate with janus room. It performs 
+following tasks -
+    * Connects with the Janus instance
+    * Fetches available rooms list
+    * Creates a new `Room` object for each room in fetched rooms list
+    * Creates a new `FeedConnection` object right after attaching to the plugin
+    (`janus.plugin.videoroom`) (for entering a room)
+    * Subscribe to -
+      + existing feeds on `"joined"` event
+      + new feeds on `"event"` event
+    * Calls `ActionService` methods to 
+      + Enter/Leave a room
+      + Publish/Unpublish a feed/screen
+      + Start/Stop ignoring a feed
+      + Toggle a channel
+      + Remote join
+      + Disable/Enable audio/video for main feed
+    * Calls `FeedsService`methods to
+      + find `Feed` object using id
+      + find `"main" Feed` object
+    * Calls `DataChannelService` methods to
+      + Broadcast status information of all feeds when a data channel is established
+      + Recieve messages from a feed
+
+02. [rooms.factory.js](../src/app/components/room/rooms.factory.js) - It defines
+`Room` object containing data about video chat room and attributes from 
+*janus.plugin.videoroom.cfg*. Some of its attributes are -
+    * id
+    * description
+    * max-publishers
+    * pin-required
+ etc.
+
+03. [feed-connection.factory.js](../src/app/components/feed/feed-connection.factory.js)
+- It defines `FeedConnection` object that manages the connection of a feed to
+the Janus server. Some of the attributes provided are - 
+    * `pluginHandle` object
+    * feed role (publisher/subscriber/screen/window)
```

Comment:
Still not precise. These are the four roles: `main`, `screen`, `window` (these three are publishers, but different kinds of publishers) and `subscriber`.